pandas/io/parsers.py

```diff
@@ -612,10 +612,8 @@ def read_csv(
     # Thus, we need a flag to indicate that we need to "override"
     # the comparison to dialect values by checking if default values
     # for BOTH "delimiter" and "sep" were provided.
-    default_sep = ","
-
     if dialect is not None:
-        sep_override = delimiter is None and sep == default_sep
+        sep_override = (delimiter is None) and (sep is lib.no_default or sep == ",")
```

Comment:
is this necessary anymore because we can no easily determine if sep is set?

Reply:
Just to give my reasoning behind this change. The problem is that this piece of code treats "," as a default value even if it provided explicitly. In this pull request, I try to make `read_csv`  and `read_table` a distinction between the default value and explicitly provided parameter value (even if it's value is identical to the default one, i.e. ",", "\t"). This statement wasn't making that distinction in the first place and there was no reason to change that. That's why I kept old behavior by complicating this if-statement. In case you think it I should revert this change and make `sep_override = False` when the function is being called like `read_csv(sep=","...)`, I can do it.

Though I can't answer your question, I hope my explanation will help. Otherwise, could provide some more input?